### PR TITLE
Add ruby 2.3.0

### PR DESCRIPTION
**Pull request by avsej**

### CODE REVIEW

File: components/ruby/ruby-23/patches/01-ruby-path-in-man-page.patch [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
++++ ruby-2.3.0/man/ruby.1	2016-04-12 12:44:33.591950380 +0300
+@@ -203,9 +203,9 @@
+ .Li #!
+ on machines that don't support it, in the following manner:
+ .Bd -literal -offset indent
+-#! /usr/local/bin/ruby
++#! /usr/ruby/2.2/bin/ruby
+ # This line makes the next one a comment in Ruby \e
+-  exec /usr/local/bin/ruby -S $0 $*
++  exec /usr/ruby/2.2/bin/ruby -S $0 $*
```
Comment (pyhalov) [inline components/ruby/ruby-23/patches/01-ruby-path-in-man-page.patch]: 2.3?

File: components/ruby/ruby-23/patches/01-ruby-path-in-man-page.patch [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ .Ed
+ .Pp
+ .It Fl T Ns Op Ar level=1
+@@ -328,7 +328,7 @@
+ .Li ARGV
+ and set the corresponding variable in the script.  For example:
+ .Bd -literal -offset indent
+-#! /usr/local/bin/ruby -s
++#! /usr/ruby/2.2/bin/ruby -s
```
Comment (pyhalov) [inline components/ruby/ruby-23/patches/01-ruby-path-in-man-page.patch]: 2.3

File: components/ruby/ruby-23/ruby-23-tk.p5m [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2015 Alexander Pyhalov
```
Comment (pyhalov) [inline components/ruby/ruby-23/ruby-23-tk.p5m]: change copyright

File: components/ruby/ruby-23/ruby-23-tk.p5m [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+file path=usr/ruby/2.3/lib/ruby/2.3.0/tkfont.rb
+file path=usr/ruby/2.3/lib/ruby/2.3.0/tkmacpkg.rb
+file path=usr/ruby/2.3/lib/ruby/2.3.0/tkmenubar.rb
+file path=usr/ruby/2.3/lib/ruby/2.3.0/tkmngfocus.rb
+file path=usr/ruby/2.3/lib/ruby/2.3.0/tkpalette.rb
+file path=usr/ruby/2.3/lib/ruby/2.3.0/tkscrollbox.rb
+file path=usr/ruby/2.3/lib/ruby/2.3.0/tktext.rb
+file path=usr/ruby/2.3/lib/ruby/2.3.0/tkvirtevent.rb
+file path=usr/ruby/2.3/lib/ruby/2.3.0/tkwinpkg.rb
+license COPYING license="Ruby License"
```
Comment (pyhalov) [inline components/ruby/ruby-23/ruby-23-tk.p5m:367]: as for ruby-23.p5m, 
a) fix copyright
b) we don't have to depend on ruby 18 anymore as it's marked obsolete

Comment (avsej) [inline components/ruby/ruby-23/ruby-23-tk.p5m:367]: fixed

**Comment by pyhalov** [comment]
Couldn't find 'usr/ruby/2.3/share/ri/2.3.0/system/ruby-2.3.0/page-README_ja_md.ri' in any of the specified search directories:
        /export/home/alp/srcs/tests/oi-userland/components/ruby/ruby-23
        /export/home/alp/srcs/tests/oi-userland/components/ruby/ruby-23/build
        /export/home/alp/srcs/tests/oi-userland/components/ruby/ruby-23/build/prototype/i386
        /export/home/alp/srcs/tests/oi-userland/components/ruby/ruby-23/build/prototype/i386/mangled
        /export/home/alp/srcs/tests/oi-userland/components/ruby/ruby-23/ruby-2.3.0
Couldn't find 'usr/ruby/2.3/share/ri/2.3.0/system/ruby-2.3.0/doc/page-ChangeLog-0_06_to_0_52.ri' in any of the specified search directories:
        /export/home/alp/srcs/tests/oi-userland/components/ruby/ruby-23
        /export/home/alp/srcs/tests/oi-userland/components/ruby/ruby-23/build
        /export/home/alp/srcs/tests/oi-userland/components/ruby/ruby-23/build/prototype/i386
        /export/home/alp/srcs/tests/oi-userland/components/ruby/ruby-23/build/prototype/i386/mangled
        /export/home/alp/srcs/tests/oi-userland/components/ruby/ruby-23/ruby-2.3.0
Couldn't find 'usr/ruby/2.3/share/ri/2.3.0/system/ruby-2.3.0/doc/page-ChangeLog-0_50_to_0_60.ri' in any of the specified search directories:
        /export/home/alp/srcs/tests/oi-userland/components/ruby/ruby-23
        /export/home/alp/srcs/tests/oi-userland/components/ruby/ruby-23/build
        /export/home/alp/srcs/tests/oi-userland/components/ruby/ruby-23/build/prototype/i386
        /export/home/alp/srcs/tests/oi-userland/components/ruby/ruby-23/build/prototype/i386/mangled
        /export/home/alp/srcs/tests/oi-userland/components/ruby/ruby-23/ruby-2.3.0
/export/home/alp/srcs/tests/oi-userland/make-rules/ips.mk:349: recipe for target '/export/home/alp/srcs/tests/oi-userland/components/ruby/ruby-23/build/manifest-i386-ruby-23.depend' failed

**Pull request merged**